internal/cache/controller_obj_cache.go

```diff
@@ -3477,6 +3511,18 @@ func ConfigureSeGroupLabels(client *clients.AviClient, seGroup *models.ServiceEn
 				utils.AviLog.Debugf("Switching to admin context from  %s", lib.GetTenant())
 				client = SharedAVIClients(lib.GetAdminTenant()).AviClient[0]
 				err := lib.AviPut(client, uri, seGroup, response)
+				if err != nil {
+					if aviError, ok := err.(session.AviError); ok && aviError.HttpStatusCode == 412 {
+						err := RefreshSeGroupDataAndCheckLabel(client, seGroup)
+						if err != nil {
+							return fmt.Errorf("Setting labels on Service Engine Group :%v failed with error :%v. Expected Labels: %v", segName, err.Error(), utils.Stringify(lib.GetLabels()))
+						}
+					} else {
+						return fmt.Errorf("Setting labels on Service Engine Group :%v failed with error :%v. Expected Labels: %v", segName, err.Error(), utils.Stringify(lib.GetLabels()))
+					}
+				}
+			} else if aviError, ok := err.(session.AviError); ok && aviError.HttpStatusCode == 412 {
```

Comment:
We don't need to type cast again, can directly use - if aviError.HttpStatusCode == 412 {}